File: CastIt.Domain/Dtos/Requests/SetFileOptionsRequestDto.cs
﻿namespace CastIt.Domain.Dtos.Requests
{
    public class SetFileOptionsRequestDto
    {
        public int StreamIndex { get; set; }
        public bool IsAudio { get; set; }
        public bool IsSubTitle { get; set; }
        public bool IsQuality { get; set; }
    }
}

File: CastIt.Android/lib/domain/models/dtos/requests/set_file_options_request_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'set_file_options_request_dto.g.dart';

@JsonSerializable()
class SetFileOptionsRequestDto {
  final int streamIndex;
  final bool isAudio;
  final bool isSubTitle;
  final bool isQuality;

  SetFileOptionsRequestDto({
    required this.streamIndex,
    required this.isAudio,
    required this.isSubTitle,
    required this.isQuality,
  });

  factory SetFileOptionsRequestDto.fromJson(Map<String, dynamic> json) => _$SetFileOptionsRequestDtoFromJson(json);

  Map<String, dynamic> toJson() => _$SetFileOptionsRequestDtoToJson(this);
}
